Normative: Create global properties for sloppy function hosting as vars

littledan opened the pull request:
This patch removes a case where a synthetic *undefined* was visible
as a result of legacy sloppy-mode function hoisting.

This patch addresses https://github.com/tc39/ecma262/issues/753

anba commented:
I wondered why we decided to use CanDeclareGlobalFunction/CreateGlobalFunctionBinding instead of CanDeclareGlobalVar/CreateGlobalVarBinding, and look what I found -> https://bugs.ecmascript.org/show_bug.cgi?id=4428#c1. 😄 

I think @syg should take a look at this change to confirm that this matches what he implemented in SpiderMonkey.

syg commented:
I'm pretty sure the PR matches my implementation. Sorry I completely forgot to respond to this.

littledan commented:
I think TC39 was happy with this sort of thing ultimately, right?

littledan commented:
I believe we [got consensus](https://github.com/tc39/tc39-notes/blob/master/es8/2017-09/sep-26.md#12ik-sloppy-function-hoisting-web-reality-tweak) here. Aside from addressing code review feedback on the tests, does anything else need to be fixed up for this patch to land?
